vendor/github.com/seccomp/libseccomp-golang/seccomp.go

```diff
@@ -20,13 +20,35 @@ import (
 
 // C wrapping code
 
-// #cgo LDFLAGS: -lseccomp
+// #cgo pkg-config: libseccomp
```

Comment:
This will still give a couple of lines with warnings that say `error`, when running update-bazel.sh.

```bash
> ./hack/update-bazel.sh                      
~/go/src/k8s.io/kubernetes/vendor ~/go/src/k8s.io/kubernetes
~/go/src/k8s.io/kubernetes
gazelle: /home/odin/go/src/k8s.io/kubernetes/vendor/github.com/seccomp/libseccomp-golang/seccomp.go: error reading go file: /home/odin/go/src/k8s.io/kubernetes/vendor/github.com/seccomp/libseccomp-golang/seccomp.go: pkg-config not supported: #cgo pkg-config: libseccomp
gazelle: /home/odin/go/src/k8s.io/kubernetes/vendor/github.com/seccomp/libseccomp-golang/seccomp_internal.go: error reading go file: /home/odin/go/src/k8s.io/kubernetes/vendor/github.com/seccomp/libseccomp-golang/seccomp_internal.go: pkg-config not supported: #cgo pkg-config: libseccomp
```

I think that it may look like errors for people without the insight into this.. Should we filter them out (hide them from the user), or should we just leave them for now? I am leaning towards filtering it out, since it annoys me quite a bit...

Reply:
I'm not sure what we should do here tbh. This is a bit annoying but the warnings are legitimate, it might be ok if we specifically filtered the warning for this package since we have a work around, but it probably would be a bad idea to generally hide them from all packages... 🤔

Reply:
Yes, it feel a bit hacky, but since we already do some funky stuff with the ldflags, i guess filtering out this warning/error is reasonable atm. Doing something like this to explicitly filter out warnings from the two files:

```bash
$ gazelle fix "...." 2>&1 | grep -Ev "vendor/github.com/seccomp/libseccomp-golang/seccomp(_internal)?.go: pkg-config not supported: #cgo pkg-config: libseccomp"
```

The PR in libseccomp-golang introducting this, https://github.com/seccomp/libseccomp-golang/pull/8, only says _Not all distributions package libseccomp in the same way, but pkg-config
allows the same configuration to work on different distributions_, but we have never had any problems with this (the old one), so it looks like all distros use the same ldflag. Maybe we can ask them to switch back? I agree that `pkg-config` is nice when dealing with different operating systems, or with more advanced stuff that require a lot of ld- and libflags, but this code will only run on linux, and is quite simple with its single ldflag... :thinking:

Reply:
Yeah. We could certainly try asking them to switch, but from their POV the `pkg-config` should technically be more correct and they probably aren't using bazel, I wouldn't bet on it... 🤔 

I'll let the other reviewers decide on the filtering or let that be another PR, I'd be happy to add it but I'm not sure if we should and it's not strictly necessary to get this fix 😅